pkg/pipeline/kafka_end_to_end_test.go

```diff
@@ -0,0 +1,141 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package pipeline
+
+import (
+	"testing"
+	"time"
+
+	"github.com/netobserv/flowlogs-pipeline/pkg/test"
+	kafkago "github.com/segmentio/kafka-go"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/net/context"
+)
+
+const kafkaConfigTemplate = `---
+log-level: debug
+pipeline:
+  - name: kafka_ingest
+  - name: decode_json
+    follows: kafka_ingest
+  - name: transform_generic
+    follows: decode_json
+  - name: kafka_encode
+    follows: transform_generic
+parameters:
+  - name: kafka_ingest
+    ingest:
+      type: kafka
+      kafka:
+        brokers: ["localhost:9092"]
+        topic: topic_in
+        groupid: group_test_in
+  - name: decode_json
+    decode:
+      type: json
+  - name: transform_generic
+    transform:
+      type: generic
+      generic:
+        rules:
+          - input: Bytes
+            output: k_bytes
+          - input: DstAddr
+            output: k_dstAddr
+          - input: DstPort
+            output: k_dstPort
+          - input: Packets
+            output: k_packets
+          - input: SrcAddr
+            output: k_srcAddr
+          - input: SrcPort
+            output: k_srcPort
+  - name: kafka_encode
+    encode:
+      type: kafka
+      kafka:
+        address: "localhost:9092"
+        topic: topic_out
+`
+
+func Test_KafkaPipeline(t *testing.T) {
+	var mainPipeline *Pipeline
+	var err error
+	v := test.InitConfig(t, kafkaConfigTemplate)
+	require.NotNil(t, v)
+
+	mainPipeline, err = NewPipeline()
+	require.NoError(t, err)
+
+	// run the pipeline in a separate go-routine
+	go func() {
+		mainPipeline.Run()
+	}()
+
+	// wait for the pipeline to come up
+	time.Sleep(time.Second)
+
+	inputData := [][]byte{
+		[]byte("{\"Bytes\":20801,\"DstAddr\":\"10.130.2.1\",\"DstPort\":36936,\"Packets\":401,\"SrcAddr\":\"10.130.2.13\",\"SrcPort\":3100}"),
+		[]byte("{\"Bytes\":20802,\"DstAddr\":\"10.130.2.2\",\"DstPort\":36936,\"Packets\":402,\"SrcAddr\":\"10.130.2.13\",\"SrcPort\":3100}"),
+		[]byte("{\"Bytes\":20803,\"DstAddr\":\"10.130.2.3\",\"DstPort\":36936,\"Packets\":403,\"SrcAddr\":\"10.130.2.13\",\"SrcPort\":3100}"),
+	}
+	expectedOutputData := [][]byte{
+		[]byte("{\"k_bytes\":20801,\"k_dstAddr\":\"10.130.2.1\",\"k_dstPort\":36936,\"k_packets\":401,\"k_srcAddr\":\"10.130.2.13\",\"k_srcPort\":3100}"),
+		[]byte("{\"k_bytes\":20802,\"k_dstAddr\":\"10.130.2.2\",\"k_dstPort\":36936,\"k_packets\":402,\"k_srcAddr\":\"10.130.2.13\",\"k_srcPort\":3100}"),
+		[]byte("{\"k_bytes\":20803,\"k_dstAddr\":\"10.130.2.3\",\"k_dstPort\":36936,\"k_packets\":403,\"k_srcAddr\":\"10.130.2.13\",\"k_srcPort\":3100}"),
+	}
+
+	// prepare a kafka producer on topic_in
+	kafkaProducer := kafkago.Writer{
```

Comment:
Why do we need golang code for this ... can we produce using some docker container ??? (adding this to the docker-compose area instead of writing golang code ??)